app/bundles/FormBundle/Controller/ResultController.php

```diff
@@ -392,6 +394,218 @@ public function deleteAction(Request $request)
+    {
+        $formId  = $this->getFormIdFromRequest();
+        $session = $request->getSession();
+        $page    = $session->get('mautic.formresult.'.$formId.'.page', 1);
+        $flashes = [];
+
+        if (Request::METHOD_POST === $request->getMethod()) {
+            $model = $this->getModel('form.submission');
+            \assert($model instanceof SubmissionModel);
+
+            $ids           = json_decode($request->query->get('ids', '[]'), true);
+            $domainsToSave = [];
+
+            foreach ($ids as $id) {
+                /** @var Submission|null $submission */
+                $submission = $model->getEntity($id);
+                if (null === $submission) {
+                    continue;
+                }
+                if (!$this->security->hasEntityAccess('form:forms:editown', 'form:forms:editother', $submission->getCreatedBy())) {
+                    $flashes[] = $this->accessDenied(true);
+                    continue;
+                }
+                $domain = $this->getDomainFromSubmission($submission);
+                if ($domain) {
+                    $domainsToSave['*@'.$domain] = true;
+                }
+            }
+
+            if ($domainsToSave) {
+                $existing      = $this->coreParametersHelper->get('do_not_submit_emails', []);
+                $hasNewDomains = false;
+
+                foreach (array_keys($domainsToSave) as $d) {
+                    if (!in_array($d, $existing, true)) {
+                        $existing[]    = $d;
+                        $hasNewDomains = true;
+                    }
+                }
+                $configurator->mergeParameters(['do_not_submit_emails' => $existing]);
+                $configurator->write();
+
+                $this->enableDonotSubmitValidationOnUnconfiguredForms();
+
+                $flashes[] = [
+                    'type' => 'notice',
+                    'msg'  => 'mautic.form.result.markspam.batch.success',
+                ];
+            } else {
+                $flashes[] = [
+                    'type' => 'notice',
+                    'msg'  => 'mautic.form.result.markspam.batch.none',
+                ];
+            }
+        }
+
+        $viewParameters = [
+            'objectId' => $formId,
+            'page'     => $page,
+        ];
+
+        return $this->postActionRedirect(
+            [
+                'returnUrl'       => $this->generateUrl('mautic_form_results', $viewParameters),
+                'viewParameters'  => $viewParameters,
+                'contentTemplate' => 'Mautic\\FormBundle\\Controller\\ResultController::indexAction',
+                'passthroughVars' => [
+                    'mauticContent' => 'formresult',
+                ],
+                'flashes' => $flashes,
+            ]
+        );
+    }
+
+    private function getDomainFromSubmission(Submission $submission): ?string
```

Comment:
I'd renamed it to 
```suggestion
    private function getEmailDomainFromSubmission(Submission $submission): ?string
```
As I wouldn't understand this is searching for an email field in the form result and is getting the domain from the email address.

I'd also move this method inside the Submission entity. There is no dependency needed and it definitely doesn't belong to the Controller layer.